### PR TITLE
New roll entry logic/robustness

### PR DESCRIPTION
**Pull request by yipingbenjaminliu**
modification from #1384 
test cases that I tested for:
for non self-attendance trainees:
  - AM inputs present roll, no roll is saved
  - AM inputs non-present roll, trainee can view
  - trainee submitted IndividualSlip, AM can view
  - AM submits none-present roll on top of the IndividualSlip, the present roll is changed
  - roll submission from roll entry table and seating chart also functional

for self-attendance trainees:
  - trainee inputs present roll, no roll is saved - good, but fail if audit roll exists.
  - AM inputs present roll, no roll is saved - fail
  - trainee inputs none-present roll from Personal Attendance, AM can view - good
  - AM inputs none-present roll from Personal Attendance, trainee can view - good
  - trainee changes roll from Personal Attendance, AM can view changes -  good
  - AM changes roll from Personal Attendance, trainee can view changes - good
  - trainee inputs roll from Personal Attendance, AM can input audit roll - good
  - AM inputs audit roll, trainee can input roll from Personal Attendance - good, see above
  - AM changes audit roll, trainee roll in Personal Attendance doesn't change - good
  - trainee changes roll from Personal Attendance, audit roll untouched. - good

Basically introduced logic flow definition in views with a slightly adjusted update method in the serializer, no need to redefine create in the serializer, logic flow in view is dictated by the url (either from personal attendance or from rolls table)

### CODE REVIEW

**Comment by yipingbenjaminliu** [comment]
this would cause the rolls that AMs input through Personal Attendance to be non-audit rolls. It might be a good idea to set it so that TAs input would also be non-audit rolls

**Comment by iamsitting** [comment]
I'll look at this on Saturday.

**Comment by iamsitting** [comment]
I'm gonna take a look this soon.

**Comment by iamsitting** [comment]
1. If self-attendance roll exists, either:
   a. Cannot submit audit roll, or
   b. Roll table does not display audit roll

**Comment by yipingbenjaminliu** [comment]
@iamsitting, I couldn't replicate the first scenario of your test. The second scenario is patched.

**Comment by yipingbenjaminliu** [comment]
@iamsitting, ready for review

**Comment by yipingbenjaminliu** [comment]
closing due to change to single roll attendance system
